packages/template-retail-react-app/app/components/product-view/variation-attribute-swatch-group.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import {useIntl} from 'react-intl'
+import PropTypes from 'prop-types'
+
+import SwatchGroup from '@salesforce/retail-react-app/app/components/swatch-group'
+import Swatch from '@salesforce/retail-react-app/app/components/swatch-group/swatch'
+import {Box} from '@salesforce/retail-react-app/app/components/shared/ui'
+
+const VariationAttributeSwatchGroup = ({id, name, selectedValue, values = [], variant}) => {
+    const intl = useIntl()
+    const swatches = values.map(({href, name, image, value, orderable}) => {
+        /** Mimic the behavior of native radio inputs by using arrow keys to select prev/next value. */
+        const onKeyDown = (evt) => {
```

Comment:
why don't we put this as a prop for the Swatch-group component letting user to choose to enable/disable this behavior? if we have it here, it looks like it is only being used by ProductView

Reply:
One goal of this PR is to update the `SwatchGroup` component so that it *does not* modify its children. (i.e. it just renders `{children}` without changing the props.)

My solution here is definitely _not_ the proper React solution, but I'm not sure the best way to implement this. The two main constraints are:
1. (as mentioned) `SwatchGroup` renders its children _without modification_
2. Using arrow keys when a swatch is focused changes the selected swatch.

Reply:
Good call. I've moved the logic back into the `Swatch` and deleted the wrapper component.